com/win32com/src/oleargs.cpp

```diff
@@ -278,9 +279,23 @@ BOOL PyCom_VariantFromPyObject(PyObject *obj, VARIANT *var)
     // So make sure this check is after anything else which qualifies.
     else if (PySequence_Check(obj)) {
         V_ARRAY(var) = NULL;  // not a valid, existing array.
-        if (!PyCom_SAFEARRAYFromPyObject(obj, &V_ARRAY(var)))
-            return FALSE;
-        V_VT(var) = VT_ARRAY | VT_VARIANT;
+        BOOL is_record_item = false;
+        if (PyObject_Length(obj) > 0) {
+            PyObject *obItemCheck = PySequence_GetItem(obj, 0);
+            is_record_item = PyRecord_Check(obItemCheck);
+        }
+        // If the sequence elements are PyRecord objects we do NOT package
+        // them as VARIANT elements but put them directly into the SAFEARRAY.
```

Comment:
I'm mildly surprised this check doesn't need to be deeper in `PyCom_SAFEARRAYFromPyObject` - are nested arrays of these a thing?  But this still looks fine for now.

Reply:
Valid point. To be honest I didn't open this Pandora's box because my use case doesn't require nested arrays.

Not sure how common they are.
I'm dealing with an engineering application that publishes round about 2000 COM methods and while the majority requires [in, out] Record pointers only 2 of them (unfortunately quite important ones) require plain Safearrays of Recods as parameters.